components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/ConfigureGradleEnterprise.java

```diff
@@ -0,0 +1,129 @@
+package com.gradle;
+
+import com.gradle.maven.extension.api.GradleEnterpriseApi;
+import com.gradle.maven.extension.api.GradleEnterpriseListener;
+import com.gradle.maven.extension.api.scan.BuildScanApi;
+import org.apache.maven.execution.MavenSession;
+import org.codehaus.plexus.logging.Logger;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import static java.lang.Boolean.parseBoolean;
+import static java.nio.file.StandardOpenOption.*;
+
+@SuppressWarnings("unused")
+public class ConfigureGradleEnterprise implements GradleEnterpriseListener {
+
+    private static final String EXPERIMENT_DIR = System.getProperty("com.gradle.enterprise.build-validation.expDir");
+    private static final String UNKNOWN_SERVER = "https://unknown.server.org";
```

Comment:
`unknown.server.org` is a viable domain name. It doesn't resolve to anything today, but in theory it could resolve to something in the future. It also means that a DNS lookup against this address will take some time (to determine that it doesn't exist). 

I'm not sure that matters too much in this case, given how we're using this currently.  But we should be careful how we use this value in the future.

Reply:
One option is `https://unknown.example.com` or anything ending with `example.com`. The `example.com` domain is reserved for this type of usage.

I think we were originally using `https://0.0.0.0`, but that's not very descriptive. It does avoid a potential DNS lookup though.

Reply:
> I had tried https://0.0.0.0 previously and it will actually still attempt to publish. 

That's surprising since we use this when we pass the flag to generate a build scan dump instead of publishing one. I imagine we would have that problem with any URL we pass here though.

Reply:
When we generate a Build Scan dump then publishing is disabled, thus nothing will not attempt to communicate with `https://0.0.0.0`.

For reference, this is what it looks like when using `https://0.0.0.0`:

![image](https://user-images.githubusercontent.com/5797900/227280686-be5b3c64-39a6-4b47-b6dd-91ac8ab196ff.png)

I may actually have a good solution though. In the `buildFinished` callback where we are validating the server, if the server is unset, then we can disable Build Scan publishing. This avoids any DNS calls, timeouts, and the dummy URL being displayed in any user facing text.

I just tried it and here is the outcome:

![image](https://user-images.githubusercontent.com/5797900/227280024-091944bf-f93e-4673-a52b-f1eef445a707.png)

Where currently it looks like:

![image](https://user-images.githubusercontent.com/5797900/227280358-0c5602f9-9b29-4a41-b3d9-45abdce315ed.png)

Reply:
I suggest we use https://unknown.example.com/. The only scenario where we run into this is when a project has the GE  extension configured but not server defined in the extension.xml.

Reply:
Can we just disable Build Scan publishing if the Gradle Enterprise server is null? Then no dummy URL is needed.

Reply:
Discussed with @etiennestuder and we decided to disable Build Scan publishing if no Gradle Enterprise is configured. This both avoids publishing to scans.gradle.com and also removes any dummy URLs.